### PR TITLE
endpoint to  the last known state of

### PR DESCRIPTION
**Pull request by simao**
- `GET /api/v1/vehicles/:vin/status`

This returns:

    {"status":"UpToDate","lastSeen":"2016-03-30T16:51:16.464+02:00"}

### CODE REVIEW

File: common-data/src/main/scala/org/genivi/sota/data/Vehicle.scala [hunk 1]
```diff
@@ -5,14 +5,16 @@
...
 
 /*
  * The notion of vehicle has a identification number (VIN), this is
  * shared between the core and resolver.
  */
-case class Vehicle(vin: Vehicle.Vin)
+case class Vehicle(vin: Vehicle.Vin, lastSeen: Option[DateTime] = None)
 
 object Vehicle {
+  def tupled: ((Vin, Option[DateTime])) ⇒ Vehicle = (Vehicle.apply _).tupled
```
Comment (jerrytrieu) [inline common-data/src/main/scala/org/genivi/sota/data/Vehicle.scala]: Plain chars: =>

Comment (simao) [inline common-data/src/main/scala/org/genivi/sota/data/Vehicle.scala]: Ops. Will update.

**Comment by jerrytrieu** [comment]
Failing test on tc

**Comment by mgarciaberlin** [comment]
```
Predicate failed: 
mr8csClWm6eer7isrdrz9aaqtyxsxciyFrfMfy8yfbfcWJu1uwjjok33xpbkkvzp1l2xwaLmlfkgvqBf35snrxpalc2odua2xregg: 
isn't a valid package name (between 1 and 100 character long alpha numeric string).
```
It was also failing on my PR, here goes a fix:
```diff
--- a/core/src/test/scala/org/genivi/sota/core/Generators.scala
+++ b/core/src/test/scala/org/genivi/sota/core/Generators.scala
@@ -31,7 +31,7 @@ trait Generators {
     Gen.listOfN(3, Gen.choose(0, 999)).map(_.mkString(".")).map(Refined.unsafeApply)
 
   val PackageNameGen: Gen[PackageId.Name] =
-    Gen.identifier.map(Refined.unsafeApply)
+    Gen.identifier.map(s => if (s.length > 100) s.substring(0, 100) else s).map(Refined.unsafeApply)
```

**Comment by mgarciaberlin** [comment]
Fix available on https://github.com/advancedtelematic/rvi_sota_server/pull/308/commits/25d162ac09bb5d630c58afe95da82d0220ea4d19

**Comment by simao** [comment]
ok, green with @mgarciaberlin's commit. Depending on which pr gets merged first, #308 or this one, I'll need to rebase with master to avoid having that commit twice.

**Comment by jerrytrieu** [comment]
Let's open new PR just for @mgarciaberlin's commit, if it fixes existing test.

**Comment by simao** [comment]
Open #312 will rebase once it's merged

**Comment by jerrytrieu** [comment]
Can rebase now.

**Pull request merged**